lib/Alchemy/Phrasea/Core/Provider/ORMServiceProvider.php

```diff
@@ -12,570 +12,207 @@
-        });
-
-        // Return configuration option for test database in DoctrineServiceProvider
-        $app['db.test.options'] = $app->share(function() use ($app) {
-            return array($app['db.test.hash.key'] => $app['db.test.info']);
-        });
-
-        // Return configuration option for test database in DoctrineServiceProvider
-        $app['db.fixture.options'] = $app->share(function() use ($app) {
-            return array($app['db.fixture.hash.key'] => $app['db.fixture.info']);
-        });
-
-        // Return configuration option for appbox database in DoctrineServiceProvider
-        $app['db.appbox.options'] = $app->share(function() use ($app) {
-            return array($app['db.appbox.hash.key'] => $app['db.appbox.info']);
-        });
-
-        // Return configuration option for databox databases in DoctrineServiceProvider
-        $app['dbs.databox.options'] = $app->share(function() use ($app) {
-            $options = array();
-
-            foreach($app['db.databox.info'] as $info) {
-                $info = $app['db.info']($info);
+            $connection = $app['dbal.provider']($connectionParameters);
 
-                $key = $app['hash.dsn']($app['db.dsn']($info));
+            $this->registerCustomTypes();
+            $this->registerEventListeners($connection->getEventManager());
 
-                $options[$key] = $info;
-            }
-
-            return $options;
-        });
-
-        // Return DoctrineServiceProvider database options, it merges all previous
-        // set database configuration
-        $app['dbs.options'] = $app->share(function() use ($app) {
-            if (false === $app['phraseanet.configuration']->isSetup()) {
-                return [];
-            }
-
-            return array_merge(
-                $app['db.appbox.options'],
-                $app['dbs.databox.options'],
-                $app['db.fixture.options'],
-                $app['db.test.options']
-            );
+            return EntityManager::create($connection, $configuration, $connection->getEventManager());
         });
 
-        // Return DoctrineORMServiceProvider information for a database from its parameters
-        $app['orm.em.options.from_info'] = $app->protect(function(array $info) use ($app) {
-            $info = $app['db.info']($info);
-
-            $key = $app['hash.dsn']($app['db.dsn']($info));
-
-            return array($key => $app['orm.options']($key));
+        $app['dbal.connection_pool'] = $app->share(function () {
+            return new ConnectionPoolManager();
         });
 
-        //Return DoctrineServiceProvider information for a database from its parameters
-        $app['db.options.from_info'] = $app->protect(function(array $info) use ($app) {
-            $info = $app['db.info']($info);
-
-            $key = $app['hash.dsn']($app['db.dsn']($info));
-
-            return array($key => $info);
+        $app['connection.pool.manager'] = $app->share(function ($app) {
```

Comment:
Yes without it you get "too many connections" errors, because the connection factory is used in a lot of places with the assumption that it always returns the same connection for a given set of parameters.